beginner_source/text_sentiment_ngrams_tutorial.py

```diff
@@ -0,0 +1,374 @@
+# 
+# ::
+# 
+#        Loss: 0.0119(train)     |       Acc: 93.6%(train)
+#        Loss: 0.0000(valid)     |       Acc: 89.6%(valid)
+#        
+# 
+# Epoch: 3 \| time in 0 minutes, 9 seconds
+# 
+# ::
+# 
+#        Loss: 0.0069(train)     |       Acc: 96.4%(train)
+#        Loss: 0.0000(valid)     |       Acc: 90.5%(valid)
+#        
+# 
+# Epoch: 4 \| time in 0 minutes, 11 seconds
+# 
+# ::
+# 
+#        Loss: 0.0038(train)     |       Acc: 98.2%(train)
+#        Loss: 0.0000(valid)     |       Acc: 90.4%(valid)
+#        
+# 
+# Epoch: 5 \| time in 0 minutes, 11 seconds
+# 
+# ::
+# 
+#        Loss: 0.0022(train)     |       Acc: 99.0%(train)
+#        Loss: 0.0000(valid)     |       Acc: 91.0%(valid)        
+# 
+
+
+######################################################################
+# Evaluate the model with test dataset
+# ------------------------------------
+# 
+
+print('Checking the results of test dataset...')
+test_loss, test_acc = test(test_dataset)
+print(f'\tLoss: {test_loss:.4f}(test)\t|\tAcc: {test_acc * 100:.1f}%(test)')
+
+
+######################################################################
+# Checking the results of test dataset…
+# 
+# ::
+# 
+#        Loss: 0.0237(test)      |       Acc: 90.5%(test)
+# 
+
+
+######################################################################
+# Test on a random news
+# ---------------------
+# 
+# Use the best model so far and test a golf news. The label information is
+# available
+# `here <https://pytorch.org/text/datasets.html?highlight=ag_news#torchtext.datasets.AG_NEWS>`__.
+# 
+
+import re
+from torchtext.data.utils import ngrams_iterator
+from torchtext.data.utils import get_tokenizer
+
+ag_news_label = {1 : "World",
+                 2 : "Sports",
+                 3 : "Business",
+                 4 : "Sci/Tec"}
+
+def predict(text, model, vocab, ngrams):
+    tokenizer = get_tokenizer("basic_english")
+    with torch.no_grad():
+        text = torch.tensor([vocab[token]
+                            for token in ngrams_iterator(tokenizer(text), ngrams)])
+        output = model(text, torch.tensor([0]))
+        return output.argmax(1).item() + 1
+
+ex_text_str = "MEMPHIS, Tenn. – Four days ago, Jon Rahm was \
+    enduring the season’s worst weather conditions on Sunday at The \
+    Open on his way to a closing 75 at Royal Portrush, which \
+    considering the wind and the rain was a respectable showing. \
+    Thursday’s first round at the WGC-FedEx St. Jude Invitational \
+    was another story. With temperatures in the mid-80s and hardly any \
+    wind, the Spaniard was 13 strokes better in a flawless round. \
+    Thanks to his best putting performance on the PGA Tour, Rahm \
+    finished with an 8-under 62 for a three-stroke lead, which \
+    was even more impressive considering he’d never played the \
+    front nine at TPC Southwind."
+
+vocab = train_dataset.get_vocab()
+with open("text_classification.pt", 'rb') as f:
```

Comment:
replace those three lines with
`
model = model.to("cpu")

print("This is a %s news" %ag_news_label[predict(ex_text_str, model, vocab, 2)])

`